x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/EsqlFeatures.java

```diff
@@ -126,6 +126,11 @@ public class EsqlFeatures implements FeatureSpecification {
      */
     public static final NodeFeature METRICS_COUNTER_FIELDS = new NodeFeature("esql.metrics_counter_fields");
 
+    /**
+     * Cast string literals to a desired data type for IN predicate and more types for BinaryComparison.
+     */
+    public static final NodeFeature STRING_LITERAL_AUTO_CASTING_8_15 = new NodeFeature("esql.string_literal_auto_casting_8_15_0");
```

Comment:
I wouldn't add the version - if anything add the date (year_month).
The former is problematic to backport while the latter isn't.

Reply:
Thank you for reviewing Costin! The last commit removed version number from the feature name.